koku/sources/koku_http_client.py

```diff
@@ -41,17 +43,91 @@ def __init__(self, auth_header):
         header = {'x-rh-identity': auth_header, 'sources-client': 'True'}
         self._identity_header = header
 
-    def create_provider(self, name, provider_type, authentication, billing_source):
-        """Koku HTTP call to create provider."""
-        url = '{}/{}/'.format(self._base_url, 'providers')
-        json_data = {'name': name, 'type': provider_type}
+    @staticmethod
+    def _get_dict_from_text_field(value):
```

Comment:
Why would private class methods need to be static?  

Should these be functions (outside of a class) instead? Or should we make them public class methods (remove the preceding `_`)?

I think either of those options is closer to best practice.

Reply:
Well these are private utility functions that's only intended to be used from this class.  They don't need any information about any particular object so there's no reference to self needed.

The purpose is to build the `/provider` HTTP json data so I don't have a use case for them outside of this class.

Reply:
If they're private functions, let's drop the `@staticmethod` decorator and call them directly from inside the class.